### PR TITLE
Added a randi() function to generate random integers

### LINKED ISSUE #123: Addition of randi() function
Hi

I would like to know the feasibility of creating a new function `randInt()` that generates a random integer. It will take an integer `value` as the argument and generate a random integer between 0 < generated <= value.

### PR DESCRIPTION
**Pull request by GhostVaibhav**
# Types of Changes

Addition of a `randi()` function to generate random integers, accepting an argument, let's say `max`. The generated number is 0 < `generated_value` <= `max`

## Prerequisites

Please make sure you can check the following two boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project

## Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [x] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

## Description

Added a function to generate random integers. This PR closes #123

### CODE REVIEW

**Comment by GhostVaibhav** [comment]
Hi @FlorianRappl, I'm not that well versed in C#, it would be great if you could have a look.

**Comment by GhostVaibhav** [comment]
The build script is not running correctly. It tries to copy files from a directory named `netstandard2.0` while the files are present in a directory named `netstandard2.1`. How should we proceed further?

File: src/Mages.Core/Runtime/Functions/StandardFunctions.cs [hunk 1]
```diff
@@ -686,6 +686,11 @@ public static class StandardFunctions
         (args.Length > 0 ? If.Is<Double>(args, SimpleRandom.CreateVector) : null) ??
         SimpleRandom.GetNumber());
 
+    /// <summary>
+    /// Contains the random integer function.
+    /// </summary>
+    public static readonly Function Randi = new(args => SimpleRandom.GetInteger((int)args[0]));
```
Comment (FlorianRappl) [inline src/Mages.Core/Runtime/Functions/StandardFunctions.cs:692]: As we don't know that `args[0]` is a double / convertable to an integer (or if we have at least 1 argument) I propose:

```cs
new(args => Curry.MinOne(Randi, args) ?? If.Is<Double>(args, x => SimpleRandom.GetInteger((int)x) ?? false)
```

**Review by FlorianRappl — COMMENTED** [review]
LGTM - only the `Randi` definition needs to consider currying and the arguments.

File: src/Mages.Core/Runtime/Functions/StandardFunctions.cs (hunk 1 above)
Comment (GhostVaibhav) [reply src/Mages.Core/Runtime/Functions/StandardFunctions.cs:692]: Just a newbie comment, but why are we checking for Double in the `If.Is<Double>` when we're actually looking for an integer?

**Comment by FlorianRappl** [comment]
> The build script is not running correctly. It tries to copy files from a directory named netstandard2.0 while the files are present in a directory named netstandard2.1. How should we proceed further?

It is broken for a longer time - I had no time to fix it until now. The problem is a bit more complex than that - there are some things which require one framework but won't work in another etc. - to fix it certain dependencies need to be replaced.

File: src/Mages.Core/Runtime/Functions/StandardFunctions.cs (hunk 1 above)
Comment (FlorianRappl) [reply src/Mages.Core/Runtime/Functions/StandardFunctions.cs]: ´There is no integer in Mages.

See the documentation: https://github.com/FlorianRappl/Mages/blob/devel/doc/types.md

Comment (GhostVaibhav) [reply src/Mages.Core/Runtime/Functions/StandardFunctions.cs]: Oh ok, gotcha 🫡

**Review by FlorianRappl — APPROVED** [review]

**Comment by GhostVaibhav** [comment]
Hi @FlorianRappl, can we backport this feature into a v2.x release? If yes, what could be the timeline for the same? If no, how should we proceed further? This PR is associated with an issue in the Powertoys repo as well linked [here](https://github.com/microsoft/PowerToys/issues/17301).

**Comment by FlorianRappl** [comment]
Theoretically yes but the build failure is not related to v3, but rather to the build agent. As such there won't be a successful build with V2 either. 

I can have a look but it will take some time as this project is currently not sponsored and therefore without priority.

**Comment by GhostVaibhav** [comment]
Surely @FlorianRappl, can you provide some rough ETA for the same?

**Pull request merged**